package simnow_logic; message DepthMarket {
  required bytes  trading_day = 1;
  required bytes  symbol = 2;
  required bytes  exchange_id = 3;
  required bytes  exchange_symbol = 4;
  required double  last_price = 5;
  required double  pre_settlement_price = 6;
  required double  pre_close_price = 7;
  required double  pre_open_interest = 8;
  required double  open_price = 9;
  required double  highest_price = 10;
  required double  lowest_price = 11;
  required double  volume = 12;
  required double  turnover = 13;
  required double  open_interest = 14;
  required double  close_price = 15;
  required double  settlement_price = 16;
  required double  upper_limit_price = 17;
  required double  lower_limit_price = 18;
  required double  pre_delta = 19;
  required double  curr_delta = 20;
  required bytes   update_time = 21;
  required int32   update_millisec = 22;
  required double  average_price = 23;
  required int64   current_time = 24;
  required bytes   action_day = 25;
  required bytes   instrument_name = 26;
  required bytes   product_id = 27;
  required double  bid_price_one = 28;
  required int32   bid_volume_one = 29;
  required double  ask_price_one = 30;
  required int32   ask_volume_one = 31;
  required double  bid_price_two = 32;
  required int32   bid_volume_two = 33;
  required double  ask_price_two = 34;
  required int32   ask_volume_two = 35;
  required double  bid_price_three = 36;
  required int32   bid_volume_three = 37;
  required double  ask_price_three = 38;
  required int32   ask_volume_three = 39;
  required double  bid_price_four = 40;
  required int32   bid_volume_four = 41;
  required double  ask_price_four = 42;
  required int32   ask_volume_four = 43;
  required double  bid_price_five = 44;
  required int32   bid_volume_five = 45;
  required double  ask_price_five = 46;
  required int32   ask_volume_five = 47;
  required double  bid_price_six = 48;
  required int32   bid_volume_six = 49;
  required double  ask_price_six = 50;
  required int32   ask_volume_six = 51;
  required double  bid_price_seven = 52;
  required int32   bid_volume_seven = 53;
  required double  ask_price_seven = 54;
  required int32   ask_volume_seven = 55;
  required double  bid_price_eight = 56;
  required int32   bid_volume_eight = 57;
  required double  ask_price_eight = 58;
  required int32   ask_volume_eight = 59;
  required double  bid_price_nine = 60;
  required int32   bid_volume_nine = 61;
  required double  ask_price_nine = 62;
  required int32   ask_volume_nine = 63;
  required double  bid_price_ten = 64;
  required int32   bid_volume_ten = 65;
  required double  ask_price_ten = 66;
  required int32   ask_volume_ten = 67;
}
